### DISCUSSION TITLE
List commits in branch

### DISCUSSION THREAD

**Discussion by knguyen100000010** [hub4j/github-api#1705] [General]
Do we currently support listing commits in a branch? `/commits?sha=[BRANCH_NAME]`? I remember I saw it implemented some where in the code, but now I cannot find it.

Thanks

**Comment by knguyen100000010** [comment]
Hi @bitwiseman, can you confirm?

**Comment by bitwiseman** [comment]
Do you mean like [this REST API ](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits)? 
I believe what you want is https://github.com/hub4j/github-api/blob/main/src/main/java/org/kohsuke/github/GHCommitQueryBuilder.java 
which you can get from [`GHRepository.queryCommits()`](https://github.com/hub4j/github-api/blob/d0809b94e38fc5cfa030e6888edf7fb2ae2ff4db/src/main/java/org/kohsuke/github/GHRepository.java#L2101l).

**Reply by knguyen100000010** [reply]
@bitwiseman exactly! Thanks sir! Yes, that was the class I looked at. Thank you!
